### ISSUE TITLE
Domain admin?

### ISSUE THREAD

**Issue by xXxOlivierxXx** [MicrosoftDocs/windowsserverdocs#2021]
On your sentence "The user has to be both local and domain admin to run this cmdlet."

Can you specify why domain admin credentials are needed? Assuming I don't want to restore the DKM, why would I need to run the command with domain admin credentials if the restore is not supposed to perform any changes to AD?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a58b49ca-ae39-132e-adbd-6766a8019a89
* Version Independent ID: fdf31266-bc9e-8880-0073-00cf1cad2897
* Content: [AD FS Rapid Restore Tool](https://docs.microsoft.com/en-us/windows-server/identity/ad-fs/operations/ad-fs-rapid-restore-tool#version-release-history)
* Content Source: [WindowsServerDocs/identity/ad-fs/operations/AD-FS-Rapid-Restore-Tool.md](https://github.com/MicrosoftDocs/windowsserverdocs/blob/master/WindowsServerDocs/identity/ad-fs/operations/AD-FS-Rapid-Restore-Tool.md)
* Product: **windows-server-threshold**
* GitHub Login: @billmath
* Microsoft Alias: **billmath**

**Comment by doctordns** [comment]
From my reading, the user has to be members of both groups in order to apply a configuration using `Restore-ADFS`. Since that cmdlet is going to restore objects to the AD, you'd need to be at least a Domain Administrator.

**Comment by xXxOlivierxXx** [comment]
I read that, but it still doesn't explain why Domain Administrator permissions are required. ADFS should not be restoring anything to/from AD, everything is restored locally in the ADFS server.

**Comment by doctordns** [comment]
> I read that, but it still doesn't explain why Domain Administrator permissions are required. ADFS should not be restoring anything to/from AD, everything is restored locally in the ADFS server.

I suspect there are read permissions to parts of AD that require domain admins group membership.

**Comment by michaelrnz** [comment]
When you run a restore it creates a new DKM key in the domain/Program Data/Microsoft/ADFS container. This requires domain admin rights from my understanding. Then your service account that runs ADFS has read rights to that location to make use of the appropriate key in encrypting/decrypting the data stored in the database.

Multiple keys appear to be fine side-by-side (just from experience, not from knowing the code), as long as the service account has rights to read the key and use it against your farm data you should be good.

**Comment by xXxOlivierxXx** [comment]
That is true when you have automatically created token signing and decrypting certificates, because the certificate key is stored in the DKM container. However, if you use third party certificates for signing and decrypting, you don't need to make use of the DKM container. In fact, Rapid Restore even has a "BackupDKM" and "RestoreDKM" parameter were you can specify if you want or not to backup/restore the DKM. If you don't specify this parameter then the Rapid restore tool doesn't do anything with the DKM container, which means it should not force you to enter domain admin credentials.

I guess this validation was not included in the code, so regardless if you specify to restore the DKM container or not, it still forces you to use a domain admin account.

**Comment by jodyhendrix** [comment]
When you create a new ADFS farm via the UI or PowerShell, it requires you to enter Domain Administrator credentials.  Now, there is a way around this using delegated permissions, however that functionality is not working in ADFS itself (due to a regression).  Once that is fixed, we'll add support for this scenario in the Rapid Restore Tool.  Unfortunately, though, I do not have an ETA for that.

**Comment by IngridAtMicrosoft** [comment]
#please-close

**Comment by IngridAtMicrosoft** [comment]
#assign:IngridAtMicrosoft 
gh issue comment 2019 -b #assign:IngridAtMicrosoft
